internal/util/setters/setters.go

```diff
@@ -62,13 +63,35 @@ func PerformSetters(path string) error {
 			// don't replace values that aren't set - stick with the defaults as defined in the manifest
 			continue
 		}
+		if c == "core.project" {
+			projectID = v
+		}
 		fltrs = append(fltrs, &setters.PerformSetters{Name: fmt.Sprintf("gcloud.%s", c), Value: v, SetBy: "kpt"})
 	}
 
+	if projectID != "" {
+		projectNumber, err := GetProjectNumberFromProjectID(projectID)
+		if err == nil && projectNumber != "" {
+			fltrs = append(fltrs, &setters.PerformSetters{Name: "gcloud.project.projectNumber", Value: projectNumber, SetBy: "kpt"})
+		}
+	}
+
 	if len(fltrs) == 0 {
 		return nil
 	}
 
 	return kio.Pipeline{Inputs: []kio.Reader{rw}, Filters: fltrs, Outputs: []kio.Writer{rw}}.
 		Execute()
 }
+
+func GetProjectNumberFromProjectID(projectID string) (string, error) {
+	gcloudCmd := exec.Command("gcloud",
+		"projects", "describe", projectID, "--format", "value(projectNumber)")
+	b, err := gcloudCmd.Output()
+	if err != nil {
+		fmt.Printf("err: %v\n", err)
```

Comment:
this error should get printed further up, use `errors.Wrap` to wrap the error and attach a stack-trace